### PR TITLE
Complete task 9 - DELETE /api/comments/:comment_id

### PR DESCRIPTION
**Pull request by Dimeben**

### CODE REVIEW

File: __tests__/comments.test.js [hunk 1]
```diff
@@ -129,3 +129,30 @@ describe("/api/articles/:article_id/comments", () => {
...
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("404 - DELETE - sends an appropriate status and error message when a non-existent comment_id is used", () => {
+    return request(app)
+      .delete("/api/comments/3123")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Page not found");
+      });
```
Comment (nc-verity-gregory) [inline __tests__/comments.test.js:156]: Sound tests 👍

File: models/comments-model.js [hunk 2]
```diff
@@ -45,3 +45,19 @@ exports.createComments = (articleId, { username, body }) => {
       return comment;
     });
 };
+
+exports.removeComment = (commentId) => {
+  if (!Number.isInteger(Number(commentId))) {
```
Comment (nc-verity-gregory) [inline models/comments-model.js:50]: You have error handlers for this. No need to custom reject

File: models/comments-model.js [hunk 3]
```diff
@@ -45,3 +45,19 @@ exports.createComments = (articleId, { username, body }) => {
...
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  return exports
+    .selectComments(commentId)
```
Comment (nc-verity-gregory) [inline models/comments-model.js:54]: There's no need to use this. You can use the RETURNING* on your query. If there are no rows, then that would be a 404 for you to reject.

File: models/comments-model.js [hunk 4]
```diff
@@ -45,3 +45,19 @@ exports.createComments = (articleId, { username, body }) => {
...
+    .then(() => {
+      return db.query(`DELETE FROM comments WHERE comment_id = $1`, [
+        commentId,
+      ]);
+    })
+    .then(() => {
```
Comment (nc-verity-gregory) [inline models/comments-model.js:60]: What is this block doing? If you consider the point above this block would be required. But currently it is doing nothing.

**Review by nc-verity-gregory — COMMENTED** [review]
Pretty solid here, you could refactor your sql statement to avoid two promises in your model.

File: models/comments-model.js (hunk 2 above)
Comment (Dimeben) [reply models/comments-model.js:50]: Screenshot 2024-08-27 at 14 19 44

File: models/comments-model.js (hunk 3 above)
Comment (Dimeben) [reply models/comments-model.js:54]: Screenshot 2024-08-27 at 14 20 15

**Pull request merged**